pkg/cmd/readyz/readyz.go

```diff
@@ -0,0 +1,253 @@
+	servingKeyFile   string
+	clientCertFile   string
+	clientKeyFile    string
+	clientCACertFile string
+
+	clientPool *etcdcli.EtcdClientPool
+}
+
+func newReadyzOpts() *readyzOpts {
+	return &readyzOpts{
+		listenPort:     defaultListenPort,
+		dialTimeout:    defaultHTTPDialTimeout,
+		targetEndpoint: defaultEndpoint,
+	}
+}
+
+// NewReadyzCommand creates a readyz command that runs as an http-get readiness server alongside the etcd member container
+func NewReadyzCommand() *cobra.Command {
+	opts := newReadyzOpts()
+	cmd := &cobra.Command{
+		Use:   "readyz",
+		Short: "Serve the HTTP /readyz endpoint health check for an etcd member",
+		Run: func(cmd *cobra.Command, args []string) {
+			defer klog.Flush()
+
+			if err := opts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := opts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	opts.AddFlags(cmd)
+	return cmd
+}
+
+func (r *readyzOpts) AddFlags(cmd *cobra.Command) {
+	fs := cmd.Flags()
+	fs.Uint16Var(&r.listenPort, "listen-port", r.listenPort, "Listen on this port. Default 9980")
+	fs.DurationVar(&r.dialTimeout, "dial-timeout", r.dialTimeout, "Dial timeout for the client. Default 2s")
+	fs.StringVar(&r.targetEndpoint, "target", r.targetEndpoint, "Target endpoint to perform health check against. Default https://localhost:2379")
+	fs.StringVar(&r.servingCertFile, "serving-cert-file", r.servingCertFile, "Health probe server TLS client certificate file. (required)")
+	fs.StringVar(&r.servingKeyFile, "serving-key-file", r.servingKeyFile, "Health probe server TLS client key file. (required)")
+	fs.StringVar(&r.clientCertFile, "client-cert-file", r.clientCertFile, "Etcd TLS client certificate file. (required)")
+	fs.StringVar(&r.clientKeyFile, "client-key-file", r.clientKeyFile, "Etcd TLS client key file. (required)")
+	fs.StringVar(&r.clientCACertFile, "client-cacert-file", r.clientCACertFile, "Etcd TLS client CA certificate file. (required)")
+	// adding klog flags to tune verbosity better
+	gfs := goflag.NewFlagSet("", goflag.ExitOnError)
+	klog.InitFlags(gfs)
+	cmd.Flags().AddGoFlagSet(gfs)
+}
+
+// Validate verifies the inputs.
+func (r *readyzOpts) Validate() error {
+	if len(r.targetEndpoint) == 0 {
+		return errors.New("missing required flag: --target")
+	}
+	if len(r.servingCertFile) == 0 {
+		return errors.New("missing required flag: --serving-cert-file")
+	}
+	if len(r.servingKeyFile) == 0 {
+		return errors.New("missing required flag: --serving-key-file")
+	}
+	if len(r.clientCertFile) == 0 {
+		return errors.New("missing required flag: --client-cert-file")
+	}
+	if len(r.clientKeyFile) == 0 {
+		return errors.New("missing required flag: --client-key-file")
+	}
+	if len(r.clientCACertFile) == 0 {
+		return errors.New("missing required flag: --client-cacert-file")
+	}
+
+	return nil
+}
+
+// Run contains the logic of the readyz command which checks the health of the etcd member
+func (r *readyzOpts) Run() error {
+	clientPool := etcdcli.NewEtcdClientPool(
```

Comment:
this does not exist in 4.9, I think you have to drop down to the old client, which had another fix I made that was important to stability:

That commit looks alright:
https://github.com/openshift/cluster-etcd-operator/commit/227e91509e72c40e26557382e8150a205aea8b55